src/statistical/unigram.jl

```diff
@@ -0,0 +1,221 @@
+"""
+struct SentencePieceModel
+  vocab::Array{String,1}
+  logprob::Array{Float64,1}
+end
+
+structure, To hold vocabulary,log probability and index  
+"""
+struct SentencePieceModel
+    vocab::Array{String,1}
```

Comment:
If we are gong to be doing `in` on this a lot, then it would be better if it was a `Set{String}`
Or possibly we have a dictionary where the `vocab` is the `keys`  and the `logprob` is the values?

Reply:
i will keep it Set{String}

Reply:
@oxinabox I thinking to keep it as the set will disturbed the order of vocab?

Reply:
In the case of dict we have to maintain ids for vocab and keys as the order is important?

Reply:
@oxinabox  should I keep it  same
or change it  ?
any suggestions

Reply:
@oxinabox

Reply:
You should change it.
`in` and `findin` in an array takes `O(n)` operations where `n` is the length of the array.
That is **very** slow, given vocab is 10,000 -100,000 words (or more?), and you have to do it for every possible token in the string being tokenized.
Where-as `haskey` and `getindex` on a `Dict` is `O(1)` i.e. constant time, and is fast.

As far as I can tell having a dict with the `vocab` as keys, and the `logprob` as values is all you need.
the order doesn't seem to matter from what i can see? I might be missing something, but if it does matter then you can use a `OrderedCollections.OrderedDict`.

You can also, if you use a dictionary, stop converting the string tokens into integer ids.
And just always work with the tokens directly.

Reply:
Thanks 
Order matters because it will act as the keys for Pretrained embedding weights here the position of vocab is the key for embedding vector  in ALBERT or other Transformer based models 
thanks for the suggestion I will make all the other changes using OrderedDict.
we don't require it to handle Unicode because we always preprocessed sentences and it always gives "unk" token to Unicode. I will change to accept Unicodes as well 
thanks

Reply:
> You can also, if you use a dictionary, stop converting the string tokens into integer ids.

Did you mean `ids_from_tokens` here?
It's for Albert_model maybe I can drop it here and implement it in ALBERT(`TextAnlaysis`) what do you think ? @oxinabox 
or maybe I can go with OrderDict as you mentioned

Reply:
I think it is better to keep tokenizer more like SentencePiece APIs which offer us Ids_for_token 
function serving as keys for ALBERT embedding weights
So, it better to use OrderDict here

Reply:
> Did you mean ids_from_tokens here?

Either is viable.

I could go either way on wether to provide ids.
On the one hand it makes sense since this is where you have the embeddings loaded.
On the other hand nothing else provides it.


Another option for how to handle the dict is to do:
 the dict as:
`vocab -> (logprob, id)` as a `Dict{String, Tuple{Float64, Int}`
Or as two dicts: `vocab->id` and `vocab->logprob`
You may also need to maintained a `id2token` mapping still which would be the current `vocab` array.

What one needs to avoid ever doing is `findin` or `in` on vectors, as that is so slow.

Reply:
Thanks, 
Got your point 
I will make all changes soon.